hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -2530,6 +2533,9 @@ def draw(
             except Exception as err:
                 add_note(err, f"while generating {key[9:]!r} from {strategy!r}")
```

Comment:
Looks like `key[9:]!r` might be wrong, if `observe_as is not None` - we should use `.removeprefix()` (well, a backport / equivalent) instead of unconditionally slicing.

Reply:
we can just use `removeprefix` since we're on 3.9 right? 😄